selfdrive/car/subaru/carstate.py

```diff
@@ -51,18 +51,30 @@ def update(self, cp, cp_cam):
     ret.cruiseState.enabled = cp.vl["CruiseControl"]['Cruise_Activated'] != 0
     ret.cruiseState.available = cp.vl["CruiseControl"]['Cruise_On'] != 0
     ret.cruiseState.speed = cp_cam.vl["ES_DashStatus"]['Cruise_Set_Speed'] * CV.KPH_TO_MS
-    # EDM Impreza: 1 = mph, UDM Forester: 7 = mph
-    if cp.vl["Dash_State"]['Units'] in [1, 7]:
-      ret.cruiseState.speed *= CV.MPH_TO_KPH
+
+    # UDM Legacy: mph = 0
+    if self.car_fingerprint == CAR.LEGACY_PREGLOBAL:
```

Comment:
Is this for all pre-global or just the legacy?

Reply:
This is only for Legacy and Forester, Outback has different value for mph. The ACC speed unit values are based on data from about 10 cars from different markets around the world. So far, it looks like Subaru has made the ACC speed unit more complex than average and is using different values depending on models and markets on the same platform

Reply:
I think another option would be removing ACC speed signal from carstate for both global and preglobal until we have a clean and reliable way to get the ACC speed unit